src/renderer/views/stake/Share/ShareView.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+
+import * as RD from '@devexperts/remote-data-ts'
+import { assetAmount, assetFromString, assetToBase, baseAmount, bnOrZero } from '@thorchain/asgardex-util'
+import { pipe } from 'fp-ts/pipeable'
+import { useObservableState } from 'observable-hooks'
+import { useIntl } from 'react-intl'
+import { useParams } from 'react-router'
+
+import PoolShare from '../../../components/uielements/poolShare'
+import { ONE_BN } from '../../../const'
+import { useMidgardContext } from '../../../contexts/MidgardContext'
+import { StakeRouteParams } from '../../../routes/stake'
+import * as helpers from './ShareView.helper'
+import * as Styled from './ShareView.styles'
+
+export const ShareView: React.FC = () => {
+  const { asset: assetParam } = useParams<StakeRouteParams>()
+  const { service: midgardService } = useMidgardContext()
+
+  const intl = useIntl()
+
+  const stakeData = useObservableState(midgardService.stake.stakes$, RD.initial)
+  const poolDetailedInfo = useObservableState(midgardService.pools.poolDetailedState$, RD.initial)
+  const runePriceRatio = useObservableState(midgardService.pools.priceRatio$, ONE_BN)
+  const priceSymbol = useObservableState(midgardService.pools.selectedPricePoolAssetSymbol$, '')
+  const asset = assetFromString(assetParam.toUpperCase())
```

Comment:
How about
```
const asset = O.fromNullable(assetFromString(assetParam.toUpperCase()))
```
?

So we can add an error message if route includes invalid asset string, but not rendering anything else. We do it similar in SendView ([code 1](https://github.com/thorchain/asgardex-electron/blob/e0277323ee23fe25c347f197bcac2d986a2b100a/src/renderer/views/wallet/send/SendView.tsx#L22), [code 2](https://github.com/thorchain/asgardex-electron/blob/e0277323ee23fe25c347f197bcac2d986a2b100a/src/renderer/views/wallet/send/SendView.tsx#L66-L77). The idea behind is to provide valid `Asset` to underlaying views only (as much as we can).

Reply:
in general i agree but.... :) there are 2 moments:
1. im not sure an Option makes sense here. we need the only check for valid data right here. 
2. we have [same check at the StakeView](https://github.com/thorchain/asgardex-electron/blob/master/src/renderer/views/stake/StakeView.tsx#L37). ShareView will never render as StakeView will return `null` right now (we should add some additional view for invalid asset. thats for sure)

Reply:
to 2) If we have already parsed `Asset` from the route and have same check before, why not just pass `Asset` directly to `ShareView`? Parsing an `Asset` from (same)  route is still unsafe and does not make sense then.